fmlloader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/AbstractModProvider.java

```diff
@@ -73,10 +74,21 @@ protected IModLocator.ModFileOrException createMod(Path path, boolean ignoreUnkn
         } else
             return new IModLocator.ModFileOrException(null, new ModFileLoadingException("Invalid mod file found " + path));
 
-        mjm.setModFile(mod);
         return new IModLocator.ModFileOrException(mod, null);
     }
 
+    private static JarMetadata loadMetaFromJar(SecureJar jar, ModJarMetadata mjm) {
```

Comment:
Should maybe be protected for third-parties that extend AbstractModProvider?
```suggestion
    protected static JarMetadata loadMetaFromJar(SecureJar jar, ModJarMetadata mjm) {
```

Reply:
Naw, I can't see a need for someone to make their own SecureJar. If they are doing that they are bypassing createMod in which case, why extend this class?